lib/mongo/client.rb

```diff
@@ -443,7 +443,12 @@ def initialize(addresses_or_uri, options = nil)
         @srv_records = uri.srv_records
       else
         addresses = addresses_or_uri
-        @srv_records = nil
+        if addresses.any? { |addr| addr =~ /mongodb(\+srv)?:\/\// }
+          raise "Protocol should not be included in host list. Did you mean to not use an array?"
+        else
+          @srv_records = nil
```

Comment:
Can you put this line back on the original indentation level please. We generally do not nest code when nesting is not needed.